evalml/automl/automl_search.py

```diff
@@ -522,9 +527,46 @@ def search(self, data_checks="auto", show_iteration_plot=True):
 
         best_pipeline = self.rankings.iloc[0]
         best_pipeline_name = best_pipeline["pipeline_name"]
+        self._best_pipeline = self.get_pipeline(best_pipeline['id'])
+        if self._train_best_pipeline:
+            X_train = _convert_to_woodwork_structure(X)
+            y_train = _convert_to_woodwork_structure(y)
+            X_threshold_tuning = None
+            y_threshold_tuning = None
+            if self.optimize_thresholds and self.objective.is_defined_for_problem_type(ProblemTypes.BINARY) and self.objective.can_optimize_threshold:
+                X_train, X_threshold_tuning, y_train, y_threshold_tuning = train_test_split(X_train, y_train, test_size=0.2, random_state=self.random_state)
+            self._best_pipeline.fit(X_train, y_train)
+            self._best_pipeline = self._threshold_pipeline(self._best_pipeline, X_threshold_tuning, y_threshold_tuning)
         logger.info(f"Best pipeline: {best_pipeline_name}")
         logger.info(f"Best pipeline {self.objective.name}: {best_pipeline['score']:3f}")
 
+    def _threshold_pipeline(self, pipeline, X_threshold_tuning, y_threshold_tuning, fold=None):
+        """Tunes the threshold of a binary pipeline to the X and y thresholding data
+
+        Arguments:
+            pipeline (Pipeline): Pipeline instance to threshold
+            X_threshold_tuning (ww DataTable): X data to tune pipeline to
+            y_threshold_tuning (ww DataColumn): Target data to tune pipeline to
+            fold (None, int): The fold of training this belongs to in AutoMLSearch. Defaults to None
+
+        Returns:
+            Trained pipeline instance
+        """
+        if self.objective.is_defined_for_problem_type(ProblemTypes.BINARY):
+            pipeline.threshold = 0.5
+            if X_threshold_tuning:
```

Comment:
Why not `if self.optimize_thresholds and self.objective.can_optimize_threshold` here like the old code did? We do need to check those things.

Reply:
@dsherry I did this since `X_threshold_tuning` would only be defined if `self.optimize_thresholds and self.objective.can_optimize_threshold` were both True. I can change it back, but this check seemed simpler and cleaner?